crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -43,6 +43,13 @@ impl<'db> UnionBuilder<'db> {
         }
     }
 
+    /// Collapse the union to a single type: `object`.
+    fn collapse_to_object(mut self) -> Self {
+        self.elements.clear();
+        self.elements.push(Type::object(self.db));
+        self
+    }
```

Comment:
I also thought about extending the `UnionBuilder` to keep track of the fact that it represented `object`. It could then short-circuit any `.add(…)` into a no-op. Not sure if it's worth the extra complexity though, as we probably don't union with `object` a lot in reality.

Reply:
This is the sort of thing IMO best left to testing with actual benchmarks once we ideally have a broader ecosystem check that we can also benchmark.